File: bank-service/src/main/java/rs/edu/raf/service/impl/CreditTypeServiceImpl.java
package rs.edu.raf.service.impl;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;
import rs.edu.raf.dto.CreditTypeCreateDTO;
import rs.edu.raf.dto.CreditTypeDTO;
import rs.edu.raf.exceptions.CreditTypeNotFoundException;
import rs.edu.raf.mapper.CreditTypeMapper;
import rs.edu.raf.model.credit.CreditType;
import rs.edu.raf.repository.CreditTypeRepository;
import rs.edu.raf.service.CreditTypeService;

import java.util.List;

@Service
@AllArgsConstructor
public class CreditTypeServiceImpl implements CreditTypeService {
    private CreditTypeRepository creditTypeRepository;
    private CreditTypeMapper creditTypeMapper;


    @Override
    public CreditTypeDTO createCreditType(CreditTypeCreateDTO creditTypeCreateDTO) {
        return creditTypeMapper.creditTypeToCreditTypeDTO(creditTypeRepository.save(creditTypeMapper.creditTypeCreateDTOtoCreditType(creditTypeCreateDTO)));
    }

    @Override
    public List<CreditTypeDTO> getAllCreditTypes() {
        return creditTypeRepository.findAll().stream().map(creditTypeMapper::creditTypeToCreditTypeDTO).toList();
    }

    @Override
    public void deleteCreditType(Long id) {
        CreditType creditType = creditTypeRepository.findById(id)
                .orElseThrow(()->new CreditTypeNotFoundException(id));

        creditTypeRepository.delete(creditType);
    }
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/service/impl/CreditTypeServiceImpl.kt
package org.example.bankservicekotlin.service.impl

import org.example.bankservicekotlin.dto.CreditTypeCreateDTO
import org.example.bankservicekotlin.dto.CreditTypeDTO
import org.example.bankservicekotlin.exception.CreditTypeNotFoundException
import org.example.bankservicekotlin.mapper.CreditTypeMapper
import org.example.bankservicekotlin.model.credit.CreditType
import org.example.bankservicekotlin.repository.CreditTypeRepository
import org.example.bankservicekotlin.service.CreditTypeService
import org.springframework.stereotype.Service

@Service
class CreditTypeServiceImpl(
    private val creditTypeRepository: CreditTypeRepository,
    private val creditTypeMapper: CreditTypeMapper
):CreditTypeService {
    override fun createCreditType(creditTypeCreateDTO: CreditTypeCreateDTO): CreditTypeDTO {
        return creditTypeMapper.creditTypeToCreditTypeDTO(
            creditTypeRepository.save(
                creditTypeMapper.creditTypeCreateDTOtoCreditType(
                    creditTypeCreateDTO
                )
            )
        )
    }

    override fun getAllCreditTypes(): List<CreditTypeDTO> {
        return creditTypeRepository.findAll().stream().map(creditTypeMapper::creditTypeToCreditTypeDTO).toList()
    }

    override fun deleteCreditType(id: Long) {
        val creditType: CreditType = creditTypeRepository.findById(id)
            .orElseThrow { CreditTypeNotFoundException(id) }

        creditTypeRepository.delete(creditType)
    }
}